File: influx-db-dev-service/runtime/src/main/java/com/thehuginn/influx/db/dev/service/runtime/InfluxDbProducer.java
package com.thehuginn.influx.db.dev.service.runtime;

import com.influxdb.client.InfluxDBClient;
import com.influxdb.client.InfluxDBClientFactory;
import com.influxdb.client.InfluxDBClientOptions;
import com.influxdb.client.QueryApi;
import com.influxdb.client.WriteApi;
import com.influxdb.client.WriteOptions;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.context.RequestScoped;
import jakarta.enterprise.inject.Produces;
import jakarta.inject.Inject;

@ApplicationScoped
public class InfluxDbProducer {

    private InfluxDBClientOptions options;

    @Produces
    @ApplicationScoped
    public InfluxDBClient createInfluxDbClient() {
        return InfluxDBClientFactory.create(options);
    }

    @Produces
    @RequestScoped
    public QueryApi createQueryApi(InfluxDBClient client) {
        return client.getQueryApi();
    }

    @Inject
    public void setConfig(InfluxDbDevServiceConfig config) {
        options = InfluxDBClientOptions.builder()
                .url(config.connectionUrl())
                .authenticate(config.username(), config.password().toCharArray())
                .org(config.orgId())
                .bucket(config.dataBucketName())
                .build();
    }

}

File: order-service/src/main/kotlin/com/thehuginn/utils/InfluxDbProducer.kt
package com.thehuginn.utils

import com.influxdb.client.InfluxDBClient
import com.influxdb.client.InfluxDBClientFactory
import com.influxdb.client.InfluxDBClientOptions
import jakarta.enterprise.context.ApplicationScoped
import jakarta.enterprise.inject.Produces
import jakarta.inject.Inject
import org.eclipse.microprofile.config.inject.ConfigProperty

@ApplicationScoped
class InfluxDBProducer {

    @Inject
    @ConfigProperty(name = "influxdb.connectionUrl")
    lateinit var influxUrl: String

    @Inject
    @ConfigProperty(name = "influxdb.data.bucketName")
    lateinit var bucketName: String

    @Inject
    @ConfigProperty(name = "influxdb.orgId")
    lateinit var orgId: String

    @Inject
    @ConfigProperty(name = "influxdb.username")
    lateinit var username: String

    @Inject
    @ConfigProperty(name = "influxdb.password")
    lateinit var password: String

    @Produces
    @ApplicationScoped
    fun createInfluxDB(): InfluxDBClient {
        val influxDBClientOptions = InfluxDBClientOptions.builder()
            .url(influxUrl)
            .authenticate(username, password.toCharArray())
            .bucket(bucketName)
            .org(orgId)
            .build()
        val influxDB = InfluxDBClientFactory.create(influxDBClientOptions)
        return influxDB
    }
}